### DISCUSSION TITLE
Try to pin a version of a single library

### DISCUSSION THREAD

**Discussion by stefan-g** [renovatebot/renovate#20767] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I would like to pin a version of a single library. I tried it with the following config but for "myLib" was still part of the merge request. Is it possible to use "ignorePaths" and "matchPackagePatterns: *" together? What  is the best practice to achive this? Would pinDependencies be better?

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "ignorePaths": ["com.test:mylib"],
      "groupName": "all non-major dependencies",
      "groupSlug": "all-minor-patch",
      "matchPackagePatterns": [
        "*"
      ],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ]
    }
  ]
}
```

### What you tried so far.

I tried it with ignoredPath and pinDependencies

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment]
use exclude package names instead of ignore paths.

ignore paths is for ignoring files and can only be used globally and not inside package rules.

**Comment by stefan-g** [comment, marked as answer]
ok if i use excludePackageNames then the library is not part of the "Update all non-major dependencies" branch but a separate change request is created. how could i prevent that?

The config looks like this:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "excludePackageNames": ["com.test:mylib"],
      "groupName": "all non-major dependencies",
      "groupSlug": "all-minor-patch",
      "matchPackagePatterns": [
        "*"
      ],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ]
    }
  ]
}
```

**Reply by viceice** [reply]
you need a second rule for that package with `enabled=false`

**Reply by stefan-g** [reply]
Thanks, that worked.  This is the config:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "groupName": "all non-major dependencies",
      "groupSlug": "all-minor-patch",
      "matchPackagePatterns": [
        "*"
      ],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ]
    },
    {
      "matchPackageNames": ["com.test:mylib"],
      "enabled": false
    }
  ]
}
```
